app/customer.py

```diff
@@ -0,0 +1,39 @@
+from math import sqrt
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(
+        self,
+        name: str,
+        product_list: dict,
+        location: list[int, int],
+        money: int,
+        car: Car,
+    ) -> None:
+        self.name = name
+        self.product_list = product_list
+        self.location = location
+        self.money = money
+        self.car = car
+
+    def distance_to_shop(self, shop: Shop) -> float:
+        return sqrt(
+            (self.location[0] - shop.location[0]) ** 2
+            + ((self.location[1] - shop.location[1]) ** 2)
+        )
+
+    def make_purchase(self, shop: Shop) -> float:
+        bill = 0
+        for product in self.product_list.keys():
```

Comment:
```suggestion
        for product in self.product_list.:
```
`keys()` - by default